src/main/java/edu/ucar/unidata/rosetta/service/EmbeddedDerbyDbInitManager.java

```diff
@@ -0,0 +1,218 @@
+                    "(" +
+                    "id INTEGER primary key not null GENERATED ALWAYS AS IDENTITY (START WITH 1, INCREMENT BY 1), " +
+                    "rosettaHome VARCHAR(100) not null, " +
+                    "uploadDir VARCHAR(100) not null, " +
+                    "downloadDir VARCHAR(100) not null, " +
+                    "maxUpload INTEGER not null" +
+                    ")";
+            createTable(createPropertiesTable, props);
+
+            String createDataTable = "CREATE TABLE data " +
+                    "(" +
+                    "id VARCHAR(100) primary key not null, " +
+                    "platform VARCHAR(100), " +
+                    "community VARCHAR(100), " +
+                    "cfType VARCHAR(100), " +
+                    "dataFileName VARCHAR(100), " +
+                    "dataFileType VARCHAR(100), " +
+                    "positionalFileName VARCHAR(100), " +
+                    "templateFileName VARCHAR(100), " +
+                    "headerLineNumbers VARCHAR(100), " +
+                    "noHeaderLines VARCHAR(10), " +
+                    "delimiter VARCHAR(50)" +
+                    ")";
+            createTable(createDataTable, props);
+
+            // Populate properties table.
+            populatePropertiesTable(props);
+
+            // Okay, we're done.  Shut down this particular connection to the database.
+            try {
+                connection = DriverManager.getConnection(url + ";shutdown=true");
+                if (connection != null) {
+                    connection.close();
+                }
+            } catch(SQLException e) {
+                // As per the Derby docs, the shutdown commands always raise SQLExceptions. (lame!)
+                logger.info("Finished creating database. Shutting down database...");
+            }
+
+
+        } else {
+            logger.info("Nothing to do here... Database already exists.");
+        }
+    }
+
+
+    /**
+     * Shuts down the embedded derby database by de-registering the driver.
+     *
+     * @param props RosettaProperties from application.properties that may be used for database shutdown.
+     * @throws SQLException   If an SQL exceptions occurs during database shutdown.
+     */
+    public void shutdownDatabase(Properties props) throws SQLException {
+
+        // Get relevant properties.
+        String rosettaHome = props.getProperty("rosetta.home");
+        String databaseName = props.getProperty("jdbc.dbName");
+        String url = props.getProperty("jdbc.url").replaceAll("\\$\\{rosetta.home}", rosettaHome);
+        url = url.replaceAll("\\$\\{jdbc.dbName}", databaseName);
+        props.setProperty("jdbc.url", url);
+
+        // Okay, we're done.  Shut down this particular connection to the database.
+        try {
+            Connection connection = DriverManager.getConnection(url + ";shutdown=true");
+            if (connection != null) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this expression which always evaluates to "true" [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2589)